### PR TITLE
Detail viewpager added

### PR DESCRIPTION
**Pull request by sanovikov71**
Change-Id: I8650dcb6f12adc1974df336a89409668c890bfcb

### CODE REVIEW

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 1]
```diff
@@ -31,6 +24,10 @@ class DetailArtObjectActivity : AppCompatActivity() {
 
     val TAG = "DetailArtObjectActivity"
 
+    val PICK_IMAGE_REQUEST = 1
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: Add this to companion object.

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 2]
```diff
@@ -40,14 +37,26 @@ class DetailArtObjectActivity : AppCompatActivity() {
...
+        val artObjectUi = dataSource.getArtObject(artObjectId!!)
         art_object_detail_name.text = artObjectUi.name
         art_object_detail_author.text = artObjectUi.authorsNames()
         art_object_detail_description.text = artObjectUi.description
 
-        art_object_detail_image_pager.adapter = GalleryPagerAdapter(this, artObjectUi.picsUrls)
+        art_object_detail_image.loadImage(artObjectUi.picsUrls[0])
+        val photosNumber = artObjectUi.picsUrls.size
+        if (photosNumber > 1) {
+            art_object_images_number.text = photosNumber.toString() + " photos"
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: http://blog.vogella.com/2011/11/22/plurals-in-android/

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 3]
```diff
@@ -40,14 +37,26 @@ class DetailArtObjectActivity : AppCompatActivity() {
...
+        artObjectId = intent.getStringExtra(EXTRA_KEY_ART_OBJECT_DETAIL_ID)
 
-        val artObjectUi = dataSource.getArtObject(artObjectId)
+        val artObjectUi = dataSource.getArtObject(artObjectId!!)
         art_object_detail_name.text = artObjectUi.name
         art_object_detail_author.text = artObjectUi.authorsNames()
         art_object_detail_description.text = artObjectUi.description
 
-        art_object_detail_image_pager.adapter = GalleryPagerAdapter(this, artObjectUi.picsUrls)
+        art_object_detail_image.loadImage(artObjectUi.picsUrls[0])
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: This also need to be view pager

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 4]
```diff
@@ -57,35 +66,15 @@ class DetailArtObjectActivity : AppCompatActivity() {
...
-            itemView.image_preview.loadImage(images[position])
-            itemView.setOnClickListener {
-                val intent = Intent(this@DetailArtObjectActivity, ImageViewActivity::class.java)
-                intent.putExtra(ImageViewActivity.INTENT_EXTRA_IMAGE_URL, images[position])
-                startActivity(intent)
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == PICK_IMAGE_REQUEST) {
+            if (resultCode == RESULT_OK) {
+                val artObjectUi = dataSource.getArtObject(artObjectId!!)
+                val position = data?.getIntExtra(ImageViewActivity.Companion.EXTRA_CHOSEN_IMAGE, -1)
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: Why -1 is default answer?

File: app/src/main/res/layout/activity_detail.xml [hunk 5]
```diff
@@ -15,18 +15,40 @@
...
             android:layout_width="match_parent"
-            android:layout_height="350dp"
+            android:layout_height="270dp"
             android:fitsSystemWindows="true"
             app:contentScrim="?attr/colorPrimary"
             app:layout_scrollFlags="scroll|exitUntilCollapsed">
 
-            <android.support.v4.view.ViewPager
-                android:id="@+id/art_object_detail_image_pager"
+            <RelativeLayout
```
Comment (genaforvena) [inline app/src/main/res/layout/activity_detail.xml:23]: This still have to be a view pager

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt (hunk 2 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: done

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt (hunk 3 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: Not actually, you can try it or just believe me

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt (hunk 4 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: done

File: app/src/main/res/layout/activity_detail.xml (hunk 5 above)
Comment (sanovikov71) [inline app/src/main/res/layout/activity_detail.xml:23]: No, the it should be just an image with some text on it

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 6]
```diff
@@ -27,10 +20,16 @@ import javax.inject.Inject
...
     companion object {
         val EXTRA_KEY_ART_OBJECT_DETAIL_ID = "EXTRA_KEY_ART_OBJECT_DETAIL_ID"
+        val EXTRA_IMAGE_CHOSEN_IN_DETAILS = "EXTRA_IMAGE_CHOSEN_IN_DETAILS"
     }
 
     val TAG = "DetailArtObjectActivity"
 
+    val PICK_IMAGE_REQUEST = 1
+
+    private var artObjectId: String? = null
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt]: This better be local val.

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 7]
```diff
@@ -27,10 +20,16 @@ import javax.inject.Inject
...
+    private var imageChosen: Int? = null
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt:31]: Won't we lost this on rotation?

**Pull request merged**